File: typescript/easy/OrderedStream.ts
class OrderedStream {
  private stream: string[];
  private ptr: number;
  constructor(n: number) {
    this.stream = new Array<string>(n);
    this.stream.fill("");
    this.ptr = 0;
  }

  insert(idKey: number, value: string): string[] {
    const rs: string[] = [];
    this.stream[idKey - 1] = value;
    while (this.ptr < this.stream.length && this.stream[this.ptr]! != "") {
      rs.push(this.stream[this.ptr]);
      this.ptr++;
    }
    return rs;
  }
}

File: java/src/easy/OrderedStream.java
package easy;

import java.util.*;

public class OrderedStream {
    private String[] list;
    private  int cur;

    public OrderedStream(int n) {
        list = new String[n];
        cur = 0;
    }

    public List<String> insert(int idKey, String value) {
        list[idKey - 1]= value;
        List<String> rs = new ArrayList<>();
        for (; cur < list.length; cur++) {
            if (list[cur] == null||list[cur] == "") {
                break;
            }
            rs.add(list[cur]);
            list[cur]= null;
        }
        return rs;
    }
}
